platforms/ide/tooling-api/src/main/java/org/gradle/tooling/internal/consumer/parameters/BuildProgressListenerAdapter.java

```diff
@@ -1056,7 +1059,7 @@ private static Failure toFailure(InternalFailure origFailure) {
             toFailures(origFailure.getCauses()));
     }
 
-    private static List<AnnotationProcessorResult> toAnnotationProcessorResults(List<InternalAnnotationProcessorResult> protocolResults) {
+    private static List<AnnotationProcessorResult> toAnnotationProcessorResults(@Nullable List<InternalAnnotationProcessorResult> protocolResults) {
```

Comment:
🤔 Seems like a code smell to have a method that takes a collection parameter have that parameter marked `@Nullable`.  Shouldn't the client just pass an empty collection?

Reply:
I agree, but the source this is filled from is @Nullable, so to get rid of the warning I can either add it here or make sure not returning `null` there is not breaking anything :(